sdk/core/azure-core/test/ut/global_context_test.cpp

```diff
@@ -40,4 +39,12 @@ TEST(Context, ApplicationContext)
   //  The context should be cancelled
   Context appContext2 = Context::ApplicationContext;
   EXPECT_TRUE(appContext2.IsCancelled());
+
+  // Reset the cancelled state of ApplicationContext.
+  Context::ApplicationContext.Reset();
+  EXPECT_FALSE(Context::ApplicationContext.IsCancelled());
+  EXPECT_TRUE(appContext2.IsCancelled());
```

Comment:
appContext2 is unaffected because Reset abandons the set of Context objects previously associated with the root context.

That's an important part of the design for cancellation - calling Cancel on a context *cannot* be undone.

Reset doesn't undo a Cancel, it Resets the shared state of the Context